DocumentManager.js

```diff
@@ -9,21 +9,52 @@ module.exports = class {
         };
     }
 
+    /**
+   * Recursively appends docIds via pagination
+   */
+    async recurseThroughPagination(cursor, docIds) {
+        let response;
+        try {
+            response = await axios.post(
+                "https://api.dropboxapi.com/2/paper/docs/list/continue",
+                {
+                    cursor: cursor
+                },
+                {
+                    headers: this.baseHeaders
+                }
+            );
+        } catch (e) {
+            console.error(e);
+            return docIds;
+        } finally {
+            Array.prototype.push.apply(docIds, response.data.doc_ids);
+            if (response.data.has_more) {
+                await this.recurseThroughPagination(response.data.cursor.value, docIds);
+            }
+        }
+    }
+
     /**
      * Return all document IDs.
      */
     async getAll() {
         let response;
+        let docIds = [];
 
         try {
             response = await axios.post("https://api.dropboxapi.com/2/paper/docs/list", {}, {
                 headers: this.baseHeaders
             });
         } catch (e) {
             console.error(e);
-            return [];
+            return docIds;
         } finally {
-            return response.data.doc_ids;
+            Array.prototype.push.apply(docIds, response.data.doc_ids);
+            if (response.data.has_more) {
+                await this.recurseThroughPagination(response.data.cursor.value, docIds);
```

Comment:
See comment above about making things a little more explicit. Maybe we just change this to assign whatever `recurseThroughPagination()` returns to the `docIds` variable. Then it's much clearer that `recurseThroughPagination()` is just collecting more document IDs for us before returning them all for Gatsby to use. 

```suggestion
                docIds = await this.recurseThroughPagination(response.data.cursor.value, docIds);
```

Reply:
I'll do it through assignment to make the coupling crystal clear. Makes sense!